src/commands/deployWorkspaceProject/internal/DeployWorkspaceProjectSaveSettingsStep.ts

```diff
@@ -27,7 +28,7 @@ export class DeployWorkspaceProjectSaveSettingsStep extends ExecuteActivityOutpu
         const deploymentConfigurations: DeploymentConfigurationSettings[] = await dwpSettingUtilsV2.getWorkspaceDeploymentConfigurations(rootFolder) ?? [];
 
         const deploymentConfiguration: DeploymentConfigurationSettings = {
-            label: context.configurationIdx !== undefined && deploymentConfigurations?.[context.configurationIdx].label || context.containerApp?.name,
+            label: context.configurationIdx !== undefined && deploymentConfigurations?.[context.configurationIdx].label || removeCaSuffixIfExists(nonNullValueAndProp(context.containerApp, 'name')),
```

Comment:
Might be cleaner to just save the `resourceName` on the context instead of doing this kind of wonky thin with the containerApp name.

Reply:
I think it's possible that the label may need to be set but there technically is no new resource being created.  So I don't think that necessarily always solves the problem of sometimes needing to check the container app name (or whichever existing resource you'd like to use to get the name).

Reply:
I'm also fine with making the label name just the container app name by default as well if the readability is suffering too much with this.  Tradeoffs I guess 😋

Reply:
> label may need to be set but there technically is no new resource being created

I can think of two cases when this happens
1. Converting v1 to v2 settings
2. Deploying to an existing container app for the first time

The `resourceName` context value might be helpful still, but we then fallback to the container app remove suffix behavior if it doesn't exist? But honestly, I'm fine either way.

Reply:
And 3. someone accidentally deletes their settings label by editing it directly.

Seems unlikely most of the time, but I bet it will happen to someone eventually.

Reply:
How about I'll add the `resourceName` field onto the context if we end up needing this value in another place :)